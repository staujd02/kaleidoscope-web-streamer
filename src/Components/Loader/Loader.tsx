import React from 'react';
import LoadScreen from '../LoadScreen/LoadScreen';
import ConfigureButton from '../Configuration/ConfigureButton';

class Loader extends React.Component<LoaderProps, LoaderState> {

    constructor(props: any) {
        super(props);
        this.state = {
            doneLoading: false
        }
        setTimeout(this.finishedLoading, this.props.loadTime);
    }

    finishedLoading = () => {
        this.setState({
            doneLoading: true
        });
    }

    render() {
        const { doneLoading } = this.state;
        const { handleConfigureClick } = this.props;
        if (doneLoading)
            return this.props.children;
        else
            return (
                <LoadScreen>
                    <ConfigureButton buttonClickHandler={handleConfigureClick} />
                </LoadScreen>
            );
    }
}

export default Loader;