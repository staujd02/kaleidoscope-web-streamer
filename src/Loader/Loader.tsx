import React from 'react';
import LoadScreen from '../LoadScreen/LoadScreen';

class Loader extends React.Component<LoaderProps, LoaderState> {

    constructor(props: any) {
        super(props);
        this.state = {
            doneLoading: false
        }
        setTimeout(this.finishedLoading.bind(this), 5000);
    }

    finishedLoading(){
        this.setState({
            doneLoading: true
        });
    }

    render(){
        const { doneLoading } = this.state;
        const { component } = this.props; 
        return (
            <div>
                {!doneLoading && <LoadScreen />}
                {doneLoading && component}
            </div>
        );
    }
}

export default Loader;