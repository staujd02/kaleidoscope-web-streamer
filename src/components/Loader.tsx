import React from 'react';
import LoadScreen from './LoadScreen/LoadScreen';

interface LoaderProps {
    component: React.Component
}

interface LoaderState {
    doneLoading: boolean
}

class Loader extends React.Component<LoaderProps, LoaderState> {
    constructor(props: any) {
        super(props);
        this.state = {
            doneLoading: false
        }
    }

    render(){
        const { doneLoading } = this.state;
        const { component } = this.props; 
        return (
            <div>
                {!doneLoading && <LoadScreen />}
                {doneLoading && component}
            </div>
        );
    }
}

export default Loader;