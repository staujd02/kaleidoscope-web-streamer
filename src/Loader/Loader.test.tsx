import React from 'react';
import Loader from './Loader';
import renderer from 'react-test-renderer';

const Component = <label>Yelp!</label> as unknown as React.Component;

describe('The Loader', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<Loader component={Component} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});