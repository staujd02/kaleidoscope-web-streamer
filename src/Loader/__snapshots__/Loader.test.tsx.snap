// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loader renders correctly 1`] = `
<div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="kaleidoscope-2.svg"
    />
    <p>
      Starting Up The Streams...
    </p>
  </header>
</div>
`;
